app/src/main/java/org/mifos/mobile/viewModels/LoanAccountTransactionViewModel.kt

```diff
@@ -25,33 +27,35 @@ class LoanAccountTransactionViewModel @Inject constructor(private val loanReposi
     val loanUiState: LiveData<LoanUiState> get() = _loanUiState
 
     fun loadLoanAccountDetails(loanId: Long?) {
-        _loanUiState.value = LoanUiState.Loading
-        loanRepositoryImp.getLoanWithAssociations(
-            Constants.TRANSACTIONS,
-            loanId,
-        )?.observeOn(AndroidSchedulers.mainThread())
-            ?.subscribeOn(Schedulers.io())
-            ?.subscribeWith(object : DisposableObserver<LoanWithAssociations?>() {
-                override fun onComplete() {}
-                override fun onError(e: Throwable) {
-                    _loanUiState.value =
-                        LoanUiState.ShowError(R.string.loan_account_details)
-                }
+        viewModelScope.launch {
```

Comment:
Other than `viewModelScope.launch {}` we don't have any other thing from the Coroutine family, do we? 😊 

What I mean by migrating to Croutines was to get rid of RxJava in the whole process of the API call
.

Reply:
done @jawidMuhammadi !!